core/src/main/java/tc/oc/pgm/modules/StatsMatchModule.java

```diff
@@ -0,0 +1,250 @@
+package tc.oc.pgm.modules;
+
+import java.text.DecimalFormat;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import net.md_5.bungee.api.ChatColor;
+import org.bukkit.Bukkit;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.match.MatchModule;
+import tc.oc.pgm.api.match.MatchScope;
+import tc.oc.pgm.api.match.event.MatchFinishEvent;
+import tc.oc.pgm.api.match.factory.MatchModuleFactory;
+import tc.oc.pgm.api.module.exception.ModuleLoadException;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.api.player.PlayerRelation;
+import tc.oc.pgm.api.player.event.MatchPlayerDeathEvent;
+import tc.oc.pgm.events.ListenerScope;
+import tc.oc.pgm.tracker.damage.ProjectileInfo;
+import tc.oc.util.bukkit.component.Component;
+import tc.oc.util.bukkit.component.ComponentUtils;
+import tc.oc.util.bukkit.component.Components;
+import tc.oc.util.bukkit.component.types.PersonalizedText;
+import tc.oc.util.bukkit.component.types.PersonalizedTranslatable;
+import tc.oc.util.bukkit.translations.AllTranslations;
+
+@ListenerScope(MatchScope.RUNNING)
+public class StatsMatchModule implements MatchModule, Listener {
+
+  public static class Factory implements MatchModuleFactory<StatsMatchModule> {
+
+    @Override
+    public StatsMatchModule createMatchModule(Match match) throws ModuleLoadException {
+      return new StatsMatchModule(match);
+    }
+  }
+
+  private final Match match;
+  private final Map<UUID, PlayerStats> allPlayerStats = new HashMap<>();
+
+  StatsMatchModule(Match match){
+    this.match = match;
+  }
+
+  public static class PlayerStats {
+    private int kills;
+    private int deaths;
+    private int killstreak;
+    private int killstreakMax;
+    private int longestBowKill;
+
+    private void onMurder() {
+      kills++;
+      killstreak++;
+      if (killstreak > killstreakMax) killstreakMax = killstreak;
+    }
+
+    private void onDeath() {
+      deaths++;
+      killstreak = 0;
+    }
+
+    private void setLongestBowKill(double distance) {
+      if (distance > longestBowKill) {
+        longestBowKill = (int) distance;
```

Comment:
To avoid the possibility of a `0` bow kill:

```suggestion
        longestBowKill = (int) Math.ceil(distance);
```

Reply:
I am not sure why that would be needed, i already prevent any message containing bow kills from showing if they are `0`.

Reply:
It's good form.